src/crimson/osd/object_context.h

```diff
@@ -49,7 +49,7 @@ class ObjectContext : public ceph::common::intrusive_lru_base<
   using watch_key_t = std::pair<uint64_t, entity_name_t>;
   std::map<watch_key_t, seastar::shared_ptr<crimson::osd::Watch>> watchers;
 
-  ObjectContext(const hobject_t &hoid) : obs(hoid) {}
+  ObjectContext(hobject_t hoid) : obs(std::move(hoid)) {}
```

Comment:
`ObjectContextRegistry::get_cached_obc(const hobject_t &hoid)` and ``intrusive_lru::get_or_create()` still take a const reference.  so we are not able to *move* the `hoid` down to `ObjectContext()` without changing all the functions on the call stack.

Reply:
Indeed, this change eliminates only the `ObjectContext` blocker towards the copy-free initialization. We can go down the stack in a follow-up.